src/agent/CMakeLists.txt

```diff
@@ -51,8 +52,14 @@ target_link_libraries(otbr-agent PRIVATE
     otbr-common
     otbr-utils
 )
+
+target_compile_definitions(otbr-agent PRIVATE
+    $<$<BOOL:${OTBR_UNSECURE_JOIN}>:OTBR_UNSECURE_JOIN_ENABLE=1>
```

Comment:
could you please follow the style of OTBR_DBUS? There is an otbr-config interface target for definitions.